MaterialDesignThemes.Wpf/NumericUpDown.cs

```diff
@@ -0,0 +1,213 @@
+
+    #endregion DependencyProperties
+
+    #region Event : ValueChangedEvent
+    [Category("Behavior")]
+    public static readonly RoutedEvent ValueChangedEvent = EventManager.RegisterRoutedEvent(nameof(ValueChanged), RoutingStrategy.Bubble, typeof(RoutedPropertyChangedEventHandler<int>), typeof(NumericUpDown));
+
+    public event RoutedPropertyChangedEventHandler<int> ValueChanged
+    {
+        add => AddHandler(ValueChangedEvent, value);
+        remove => RemoveHandler(ValueChangedEvent, value);
+    }
+    #endregion Event : ValueChangedEvent
+
+    public override void OnApplyTemplate()
+    {
+        if (_increaseButton != null)
+            _increaseButton.Click -= IncreaseButtonOnClick;
+
+        if (_decreaseButton != null)
+            _decreaseButton.Click -= DecreaseButtonOnClick;
+        if (_textBoxField != null)
+            _textBoxField!.TextChanged -= OnTextBoxFocusLost;
+
+        _increaseButton = GetTemplateChild(IncreaseButtonPartName) as RepeatButton;
+        _decreaseButton = GetTemplateChild(DecreaseButtonPartName) as RepeatButton;
+        _textBoxField = GetTemplateChild(TextFieldBoxPartName) as TextBox;
+
+        if (_increaseButton != null)
+            _increaseButton.Click += IncreaseButtonOnClick;
+
+        if (_decreaseButton != null)
+            _decreaseButton!.Click += DecreaseButtonOnClick;
+
+        if (_textBoxField != null)
+        {
+            _textBoxField!.LostFocus += OnTextBoxFocusLost;
+            _textBoxField.Text = Value.ToString(CultureInfo.CurrentUICulture);
+        }
+
+        base.OnApplyTemplate();
+    }
+
+    private void OnTextBoxFocusLost(object sender, EventArgs e)
+    {
+        if (int.TryParse(_textBoxField?.Text, NumberStyles.Integer, CultureInfo.CurrentUICulture, out int numericValue))
+        {
+            SetCurrentValue(ValueProperty, numericValue);
+        }
+
+    }
+    private void IncreaseButtonOnClick(object sender, RoutedEventArgs e)
+    {
+        OnIncrease();
+    }
+
+    private void DecreaseButtonOnClick(object sender, RoutedEventArgs e)
+    {
+        OnDecrease();
+    }
+
+    private void OnIncrease()
+    {
+        SetCurrentValue(ValueProperty, Value + 1);
+
+        if (IncreaseCommand?.CanExecute(null) ?? false)
+            IncreaseCommand.Execute(null);
+    }
+
+    private void OnDecrease()
+    {
+        SetCurrentValue(ValueProperty, Value - 1);
+
+        if (DecreaseCommand?.CanExecute(this) ?? false)
+            DecreaseCommand.Execute(this);
+    }
+
+    protected override void OnPreviewKeyDown(KeyEventArgs e)
+    {
+        if (e.Key == Key.Up)
+        {
+            OnIncrease();
+            e.Handled = true;
+        }
+        else if (e.Key == Key.Down)
+        {
+            OnDecrease();
+            e.Handled = true;
+        }
+        base.OnPreviewKeyDown(e);
+    }
+
+    protected override void OnPreviewMouseWheel(MouseWheelEventArgs e)
+    {
+        if (IsKeyboardFocusWithin)
```

Comment:
It might be nice to create another dependency property to disable this behavior for those who don't want it.